### PR TITLE
feat(dagent): add useParent functionality to dagent

### PR DESCRIPTION
**Pull request by nandor-magyar**
It was implemented a while ago in our crane, but it was missing from dagent.

### CODE REVIEW

**Review by polaroi8d — APPROVED** [review]
LGTM (informal)

File: golang/internal/helper/image/image.go [hunk 1]
```diff
@@ -204,49 +205,80 @@ func Pull(ctx context.Context, cli client.APIClient, logger io.StringWriter, exp
...
-	cli, cliErr := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
-	if cliErr != nil {
-		return cliErr
-	}
-
-	if !forcePull {
+func shouldUseLocalImage(ctx context.Context, cli client.APIClient,
+	distributionRef reference.Named, encodedAuth string, preferLocal bool,
+) (bool, error) {
+	if preferLocal {
```
Comment (robot9706) [inline golang/internal/helper/image/image.go:225]: Feels a bit strange calling this function only to check a param, while you could check it before calling it.

Also `preferLocal` feels like it should only be considered if both local and remote images exist and then local should be used. So maybe `forceLocal` or something like that?

File: golang/pkg/dagent/utils/init_container.go [hunk 2]
```diff
@@ -45,16 +51,14 @@ func spawnInitContainer(
 
 	resultCont, waitResult, err := builder.
 		WithClient(cli).
-		WithImage(cont.Image).
-		WithCmd(cont.Command).
+		WithImage(initCont.Config.Name).
+		WithCmd(initCont.Config.Command).
 		WithName(initContName).
-		WithEnv(EnvMapToSlice(cont.Envs)).
+		WithEnv(initCont.EnvList).
```
Comment (robot9706) [inline golang/pkg/dagent/utils/init_container.go:57]: We should use both InitContainer envs + parent config envs.

**Review by robot9706 — CHANGES_REQUESTED** [review]

File: golang/pkg/dagent/utils/init_container.go (hunk 2 above)
Comment (nandor-magyar) [reply golang/pkg/dagent/utils/init_container.go:57]: Nice catch.

File: golang/internal/helper/image/image.go (hunk 1 above)
Comment (nandor-magyar) [reply golang/internal/helper/image/image.go:225]: As discussed it is prefer local because it attempts to use remote if local is not available.

**Comment by robot9706** [comment]
Tested, init containers seem to use their name as their image name, and the containers do not actually have any environment variables.

**Review by robot9706 — APPROVED** [review]
Tested, seems to be working fine 🎈

**Pull request merged**